shared/User.d.ts

```diff
@@ -1,20 +1,30 @@
 import UserStats from './UserStats';
 
 /** A base user represents someone who hasn't made an account yet, and just wants to play without going through the hassle. */
-export interface BaseUser {
+interface BaseUser {
+    /** Valid names are between 2 and 20 (inclusive) characters long. */
     name: string;
+    isGuest: boolean;
+}
+
+export interface GuestUser extends BaseUser {
+    isGuest: true;
 }
 
 /** A full user represents someone who has made an account.
  *
  * Eventually some form of email authentication would be nice here.
  */
 export interface FullUser extends BaseUser {
+    /** Name but in lowercase. */
+    nameL: string;
+    id: string;
```

Comment:
Is there a purpose in having a separate id from the mongo object id?

Reply:
We can't access the Mongoose document `_id` property unless we make the interface extend Mongoose's `Document` type, which is not recommended.